presto-main/src/test/java/com/facebook/presto/sql/planner/TestLocalExecutionPlanner.java

```diff
@@ -47,6 +47,13 @@ public void testCompilerFailure()
         assertFails("SELECT " + outer, COMPILER_ERROR);
     }
 
+    @Test
+    public void testCompiler()
+    {
+        // in-list has decimal values with different precision and cast the list to double
+        runner.execute("SELECT c1, CASE WHEN c1 IN ( DECIMAL '2.0', DECIMAL '30.0' ) THEN 'a' ELSE 'b' END FROM ( VALUES ( CAST (1 AS DOUBLE) ) ) T (c1)");
```

Comment:
Use query variant as simple as possible to test your case. I think it will have same value if you'd just do:
```
        // coercion with type only coercion inside IN list
        assertQuery("SELECT d IN (DECIMAL '2.0', DECIMAL '30.0') FROM (VALUES (DOUBLE '2.0')) t(d)", "VALUES TRUE");
```